x/params/client/rest/grpc_query_test.go

```diff
@@ -0,0 +1,128 @@
+package rest_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/cosmos/cosmos-sdk/testutil"
+	"github.com/cosmos/cosmos-sdk/testutil/network"
+	"github.com/cosmos/cosmos-sdk/x/params/types/proposal"
+	"github.com/gogo/protobuf/proto"
+	"github.com/stretchr/testify/suite"
+)
+
+type IntegrationTestSuite struct {
+	suite.Suite
+
+	cfg     network.Config
+	network *network.Network
+}
+
+func (s *IntegrationTestSuite) SetupSuite() {
+	s.T().Log("setting up integration test suite")
+
+	cfg := network.DefaultConfig()
+	cfg.NumValidators = 1
+
+	s.cfg = cfg
+	s.network = network.New(s.T(), cfg)
+
+	_, err := s.network.WaitForHeight(1)
+	s.Require().NoError(err)
+}
+
+func (s *IntegrationTestSuite) TearDownSuite() {
+	s.T().Log("tearing down integration test suite")
+	s.network.Cleanup()
+}
+
+func (s *IntegrationTestSuite) TestQueryParamsGRPC() {
+	val := s.network.Validators[0]
+	baseURL := val.APIAddress
+
+	testCases := []struct {
+		name     string
+		url      string
+		headers  map[string]string
+		expErr   bool
+		respType proto.Message
+		expected proto.Message
+	}{
+		{
+			"with no subspace, key",
+			fmt.Sprintf("%s/cosmos/params/v1beta1/params?subspace=%s&key=%s", baseURL, "", ""),
+			map[string]string{},
```

Comment:
yeah, this looks less distracting than https://github.com/cosmos/cosmos-sdk/pull/7173#discussion_r482969324 👍. You could even remove `headers  map[string]string` from the testCase struct

Reply:
We need to pass height header, response is height specific here.